food-truck-frontend/src/api/RateReview.ts

```diff
@@ -0,0 +1,201 @@
+    if (name.data !== null) {
+      // Get the reviews by this user
+      let resp: any = await api.request({
+        url: `/reviews/truck/${userId}`,
+        method: "GET",
+      });
+
+      // Map all backend reviews to frontend reviews
+      reviews = resp.data.map(async (r: any) => {
+        // Create frontend representation for 1 review, with either the username or a blank string
+        return review_backendToFrontend(r, name.data.username);
+      });
+    }
+  } catch (e: any) {
+    // Derp. Handle failure as the client requested and return an empty list
+    reviews = [];
+    if (onFail) onFail(e);
+  }
+
+  return reviews;
+};
+
+/**
+ * Get the review on a truck by a particular user
+ * @param truckId The ID of the truck to load the review for
+ * @param userId The ID of the user to load the review from
+ * @param onFail An optional response to the API request failing
+ */
+export const loadReviewFromTruckForUser = async (
+  truckId: number,
+  userId: number,
+  onFail?: ClientResponseAction
+) => {
+  let review: Review | null = null;
+  try {
+    // Fetch username
+    let name: any = await api.request({
+      url: `/user/${userId}`,
+      method: "GET",
+    });
+
+    // If we got a good response, then this user exists; create frontend representation of reviews
+    if (name.data !== null) {
+      // Get the reviews by this user
+      let resp: any = await api.request({
+        url: `/reviews/truck/${truckId}/user`,
+        params: {
+          userId: userId
+        },
+        method: "GET",
+      });
+
+      // Map review to a frontend review
+      if (resp.data !== null)
+        review = review_backendToFrontend(resp.data, name.data.username);
+    }
+  } catch (e: any) {
+    // Derp. Handle failure as the client requested and return null
+    review = null;
+    if (onFail) onFail(e);
+  }
+
+  return review;
+};
+
+/**
+ * Get all reviews for a specific truck
+ * @param truckId The ID of the truck with reviews to retrieve
+ * @param onFail An optional response to the API request failing
+ */
+export const loadReviewsByTruck = async (
+  truckId: number,
+  onFail?: ClientResponseAction
+) => {
+  let reviews: TruckReviews = {
+    reviews: [],
+    avgCostRating: null,
+    avgStarRating: null,
+    truckName: ""
+  };
+
+  try {
+    // Get the reviews for this truck
+    let resp: any = await api.request({
+      url: `/reviews/truck/${truckId}`,
+      method: "GET",
+    });
+
+    // If we got a good response, then create frontend representation
+    if (resp.data !== null) {
+      // Map all backend reviews to frontend reviews
+      // Learned to use `Promise.all` from: https://stackoverflow.com/questions/40140149/use-async-await-with-array-map
+      reviews.reviews = await Promise.all(resp.data.map(async (r: any) => {
+        // Fetch username
+        let name: any = await api.request({
+          url: `/user/${r.userId}`,
+          method: "GET",
+        });
```

Comment:
It probably would have been better if the backend returned a "ReviewView" or something that had all the info we needed on the frontend. But since we already have the work in the frontend, might as well keep it.